src/api/c++/z3++.h

```diff
@@ -2255,14 +2342,21 @@ namespace z3 {
         model(context & c):object(c) { init(Z3_mk_model(c)); }
         model(context & c, Z3_model m):object(c) { init(m); }
         model(model const & s):object(s) { init(s.m_model); }
+        model(model && s) noexcept : object{std::forward<object>(s)}, m_model{s.m_model} { s.m_model = nullptr; }
         model(model& src, context& dst, translate) : object(dst) { init(Z3_model_translate(src.ctx(), src, dst)); }
-        ~model() { Z3_model_dec_ref(ctx(), m_model); }
+        ~model() { if(m_model != nullptr) Z3_model_dec_ref(ctx(), m_model); }
```

Comment:
Z3_model_dec_ref also works with null

Reply:
I'll take a look at which `dec_ref` functions work with `nullptr`. _Should_ they all? I.e. any that do not, should I change the `dec_ref` source code? Or are they all correct, except for the one in `api_ast_vector.cpp` as you said above?